api-put-object.go

```diff
@@ -78,7 +81,7 @@ func (opts PutObjectOptions) Header() (header http.Header) {
 		header[amzHeaderMatDesc] = []string{opts.EncryptMaterials.GetDesc()}
 	}
 	for k, v := range opts.UserMetadata {
-		if !isAmzHeader(k) && !isStandardHeader(k) && !isSSEHeader(k) {
+		if !isAmzHeader(k) && !isStandardHeader(k) && !isSSEHeader(k) && isValidHeader(k) {
```

Comment:
Not a good idea to skip the invalid inputs, it can be unexpected from an API point of view - we should return error during validate() for options and return proper error.